### PR TITLE
General Refactor

### PR DESCRIPTION
**Pull request by Mzack9999**
- Updated GH actions
- Go version bump bump
- Fixing various bugs causing crashes
- Using nuclei latest version instead of outdated one

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: 🚨 CodeQL Analysis
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - dev
```
Comment (tarunKoyalwar) [inline .github/workflows/codeql-analysis.yml:7]: @Mzack9999  I think we can omit this

**Review by tarunKoyalwar — APPROVED** [review]
lgtm!

**Comment by princechaddha** [comment]
```yaml

**Comment by Mzack9999** [comment]
@princechaddha The file is automatically excluded, and only a warning is printed to stdout. I've modified the code to an informative message stating that the file is ignored.

**Review by princechaddha — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**